repose-aggregator/commons/commons-utilities/src/main/scala/org/openrepose/commons/utils/servlet/http/HttpServletResponseWrapper.scala

```diff
@@ -85,6 +87,7 @@ class HttpServletResponseWrapper(originalResponse: HttpServletResponse, headerMo
     case ResponseMode.MUTABLE => new MutableServletOutputStream(desiredOutputStream)
   }
 
+  private var statusCode: Int = originalResponse.getStatus
```

Comment:
Oh god, i understand why, but i suspect we are going to get some super weird behavior out of this.

Reply:
I hope not. It does kind of depend on the container, but between copying the wrapped response's behavior and always defaulting to our own code, I thought this was the better idea. That is, if no one has called either `setStatus` or `sendError`, we probably shouldn't modify the response code, but we should report it accurately. So this does that.